main.go

```diff
@@ -7,6 +7,7 @@ import (
 	"net/http"
 	"os"
 	"os/signal"
+	"reflect"
```

Comment:
We don't need `reflect` package for this, we can achieve the same with a core functionality [type switch](https://go.dev/tour/methods/16).

Reply:
Ok. I thought with `reflect` it looks better, but you know better 🙃